server/notification-providers/teams.js

```diff
@@ -48,6 +63,137 @@ class Teams extends NotificationProvider {
+                                                        "url": "https://raw.githubusercontent.com/louislam/uptime-kuma/master/public/icon.png",
+                                                        "altText": "Uptime Kuma Logo"
+                                                    }
+                                                ]
+                                            },
+                                            {
+                                                "type": "Column",
+                                                "width": "stretch",
+                                                "items": [
+                                                    {
+                                                        "type": "TextBlock",
+                                                        "size": "Medium",
+                                                        "weight": "Bolder",
+                                                        "text": `**${this._statusMessageFactory(status, monitorName, false)}**`,
+                                                    },
+                                                    {
+                                                        "type": "TextBlock",
+                                                        "size": "Small",
+                                                        "weight": "Default",
+                                                        "text": "Uptime Kuma Alert",
+                                                        "isSubtle": true,
+                                                        "spacing": "None"
+                                                    }
+                                                ]
+                                            }
+                                        ]
+                                    }
+                                ]
+                            },
+                            {
+                                "type": "FactSet",
+                                "separator": false,
+                                "facts": facts
+                            }
+                        ],
+                        "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
+                        "version": "1.4"
+                    }
+                }
+            ]
+        };
+
+        if (actions) {
+            payload.attachments[0].content.body.push({
+                "type": "ActionSet",
+                "actions": actions,
+            });
+        }
+
+        return payload;
+    };
+
+    /**
+     * Generate payload for notification
+     * @param {const} status The status of the monitor
+     * @param {string} monitorMessage Message to send
+     * @param {string} monitorName Name of monitor affected
+     * @param {string} monitorUrl URL of monitor affected
+     * @returns {Object}
+     */
+    _notificationPayloadFactoryLegacy = ({
```

Comment:
Please remove the legacy code (including where this function is used)
If microsoft decides to no longer support the webhooks, it is not worth adding code to support them.

Since the feature would be broken otherwise I think this is still a semver patch without this.

Currently, this just produces a bigger merge conflict ^^

Reply:
Sure, I can do that.
My thinking was, that i want to keep the old Webhooks running until Mircosoft gets rid of them.
Thought that breaking the old webhooks before that would result in this being classified as a breaking change.

Normally I would have removed that code in a second PR after this one.
If this is a hard requirement to get this merged, i get rid of it now.